inc/pkthlp.h

```diff
@@ -308,6 +309,138 @@ PktBuildTcpFrame(
     return TRUE;
 }
 
+inline
+_Success_(return != FALSE)
+BOOLEAN
+PktBuildQuicV1PacketLongHeader(
+    _Out_writes_bytes_(*BufferSize) VOID *Buffer,
+    _Inout_ UINT32 *BufferSize,
+    _In_ UINT8 TypeAndSpecificBits,
+    _In_reads_bytes_(DestConnIdLength) CONST UCHAR *DestConnId,
+    _In_ UINT8 DestConnIdLength,
+    _In_reads_bytes_(SrcConnIdLength) CONST UCHAR *SrcConnId,
+    _In_ UINT8 SrcConnIdLength,
+    _In_opt_ CONST UCHAR *Payload,
+    _In_ UINT16 PayloadLength
+    )
+{
+    UINT8* Cursor = (UINT8*)Buffer;
+    CONST UINT32 Version = 1;
+    CONST UINT32 TotalLength = sizeof(UINT8) + sizeof(Version) + sizeof(DestConnIdLength) +
+        DestConnIdLength + sizeof(SrcConnIdLength) + SrcConnIdLength + PayloadLength;
+
+    if (*BufferSize < TotalLength) {
+        return FALSE;
+    }
+
+    //
+    // Set the Header Form bit to 1 for a long header, and Fixed bit to 1
+    //
+    CONST UINT8 Prelude = 0xC0 | (0x3F & TypeAndSpecificBits);
+
+    *Cursor = Prelude;
+    Cursor += sizeof(Prelude);
+
+    RtlCopyMemory(Cursor, &Version, sizeof(Version));
+    Cursor += sizeof(Version);
+
+    *Cursor = DestConnIdLength;
+    Cursor += sizeof(DestConnIdLength);
+    RtlCopyMemory(Cursor, DestConnId, DestConnIdLength);
+    Cursor += DestConnIdLength;
+
+    *Cursor = SrcConnIdLength;
+    Cursor += sizeof(SrcConnIdLength);
+    RtlCopyMemory(Cursor, SrcConnId, SrcConnIdLength);
+    Cursor += SrcConnIdLength;
+
+    if (Payload) {
```

Comment:
```suggestion
    if (Payload != NULL) {
```

similar pattern elsewhere